test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -3457,6 +3457,62 @@ public async Task GetStatus_WithCondition(bool extendedSessions, string storageP
             }
         }
 
+        [Theory(Skip = "Azure Storage fails due to container deletion")]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [MemberData(nameof(TestDataGenerator.GetExtendedSessionAndFullFeaturedStorageProviderOptions), MemberType = typeof(TestDataGenerator))]
+        public async Task DurableEntity_GetStatus_WithCondition(bool extendedSessions, string storageProvider, bool fetchInput)
+        {
+            var taskHubName = "GetStatus1";
+            await TestHelpers.DeleteTaskHubResources(taskHubName, extendedSessions);
+            using (JobHost host = TestHelpers.GetJobHost(this.loggerProvider, taskHubName, extendedSessions, storageProviderType: storageProvider))
+            {
+                await host.StartAsync();
+
+                var id1 = new EntityId("aaab", "foo");
+                var id2 = new EntityId("aaab", "bar");
+                var id3 = new EntityId("aaab", "baz");
+                var idDifferentName = new EntityId("aaac", "foo");
+
+                var client1 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.EntityId_SignalAndCallStringStore), id1, this.output);
+                var client2 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.EntityId_SignalAndCallStringStore), id2, this.output);
+                var client3 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.EntityId_SignalAndCallStringStore), id3, this.output);
+                var client4 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.EntityId_SignalAndCallStringStore), idDifferentName, this.output);
+
+                var yesterday = DateTime.UtcNow.Subtract(TimeSpan.FromDays(1));
+                var tomorrow = DateTime.UtcNow.Add(TimeSpan.FromDays(1));
+
+                var condition1 = new EntityQuery
+                {
+                    EntityName = "aaab",
+                    LastOperationFrom = yesterday,
+                    LastOperationTo = tomorrow,
+                    FetchInput = fetchInput,
+                };
+
+                // Make sure it actually completed
+                await client1.WaitForCompletionAsync(this.output);
+                await client2.WaitForCompletionAsync(this.output);
+                await client3.WaitForCompletionAsync(this.output);
+                await client4.WaitForCompletionAsync(this.output);
+
+                // Perform some operations
+                var result = await client1.InnerClient.ListEntitiesAsync(condition1, CancellationToken.None);
+
+                Assert.Equal(3, result.Entities.Count());
+
+                if (fetchInput)
+                {
+                    Assert.NotNull(result.Entities.First());
+                }
+                else
+                {
+                    Assert.Null(result.Entities.First());
+                }
+
+                await host.StopAsync();

```

Comment:
I suggest also testing a few cases where none of the entities match the query (wrong EntityName, wrong time range, etc.).